### PR TITLE
Add options to enable/disable LAN discovery

### PR DESCRIPTION
**Pull request by kurnevsky**

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [[package]]
 name = "aho-corasick"
-version = "0.6.4"
+version = "0.6.5"
```
Comment (kpp) [inline Cargo.lock]: You may update only tox package by `cargo update -p tox`

File: src/cli_config.rs [hunk 2]
```diff
@@ -138,12 +148,15 @@ pub fn cli_parse() -> CliConfig {
 
     let motd = value_t!(matches.value_of("motd"), String).unwrap_or_else(|e| e.exit());
 
+    let lan_discovery_enabled = !matches.is_present("no-lan");
```
Comment (kpp) [inline src/cli_config.rs]: why not `let lan_discovery_enabled = matches.is_present("lan");`?

Comment (kurnevsky) [reply src/cli_config.rs]: Because if neither of these args are specified default behavior - enable lan discovery.

File: src/main.rs [hunk 3]
```diff
@@ -131,10 +131,15 @@ fn main() {
     // Create a channel for server to communicate with network
     let (tx, rx) = mpsc::unbounded();
 
-    let lan_discovery_sender = LanDiscoverySender::new(tx.clone(), dht_pk, udp_addr.is_ipv6());
+    let lan_discovery_future = if cli_config.lan_discovery_enabled {
+        LanDiscoverySender::new(tx.clone(), dht_pk, udp_addr.is_ipv6()).run()
+    } else {
+        Box::new(future::empty())
```
Comment (kpp) [inline src/main.rs]: You combine futures like this: `future.select(lan_discovery_future)`. Shouldn't `future::empty()` resolve as fast as possible?

File: Cargo.lock (hunk 1 above)
Comment (kurnevsky) [reply Cargo.lock]: Does it make sense? Since everything is working fine with updated versions.

File: src/main.rs (hunk 3 above)
Comment (kurnevsky) [reply src/main.rs]: Empty is a future that never resolves.

File: src/cli_config.rs (hunk 2 above)
Comment (kpp) [reply src/cli_config.rs]: Maybe better use only one flag like `--no-lan`?

Comment (kpp) [reply src/cli_config.rs]: Or `--lan=[enabled/disabled]` with default value = `enabled`

File: src/main.rs (hunk 3 above)
Comment (kpp) [reply src/main.rs]: Thanks.

File: src/cli_config.rs (hunk 2 above)
Comment (kurnevsky) [reply src/cli_config.rs]: I'd prefer just flag - it's simpler.

Comment (kpp) [reply src/cli_config.rs]: I would prefer `--lan=[enabled/disabled]`, that gives more info about the default behaviour.

**Pull request merged**